crates/ruff_linter/src/rules/flake8_bugbear/rules/useless_comparison.rs

```diff
@@ -33,33 +34,64 @@ use super::super::helpers::at_last_top_level_expression_in_cell;
 /// ## References
 /// - [Python documentation: `assert` statement](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement)
 #[violation]
-pub struct UselessComparison;
+pub struct UselessComparison {
+    at: ComparisonLocationAt,
+}
 
 impl Violation for UselessComparison {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!(
-            "Pointless comparison. Did you mean to assign a value? \
-             Otherwise, prepend `assert` or remove it."
-        )
+        match self.at {
+            ComparisonLocationAt::MiddleBody => format!(
+                "Pointless comparison. Did you mean to assign a value? \
+                Otherwise, prepend `assert` or remove it."
+            ),
+            ComparisonLocationAt::EndOfFunction => format!(
+                "Pointless comparison at end of function scope. Did you mean \
+                to return the expression result?"
+            ),
+        }
     }
 }
 
 /// B015
 pub(crate) fn useless_comparison(checker: &mut Checker, expr: &Expr) {
     if expr.is_compare_expr() {
+        let semantic = checker.semantic();
+
         if checker.source_type.is_ipynb()
             && at_last_top_level_expression_in_cell(
-                checker.semantic(),
+                semantic,
                 checker.locator(),
                 checker.cell_offsets(),
             )
         {
             return;
         }
 
-        checker
-            .diagnostics
-            .push(Diagnostic::new(UselessComparison, expr.range()));
+        if let ScopeKind::Function(func_def) = semantic.current_scope().kind {
+            if func_def.range.end() == expr.range().end() {
```

Comment:
I think this works but it's a fairly subtle assumption that this will always hold true. We actually considered changing the parser to include the range of the `decent` in the range of the function (which would simplify the formatter by a lot). 

That's why I would prefer a check like this
```suggestion
						let is_last = function
                .body
                .last()
                .and_then(Stmt::as_expr_stmt)
                .is_some_and(|last_stmt| &*last_stmt.value == expr);
            if is_last {
```

Reply:
Thank you for the suggestion!